src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int countObjects = 5;
+
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] arrayOfFigures = new Figure[countObjects];
+        for (int i = 0; i < arrayOfFigures.length; i++) {
+            if (i < arrayOfFigures.length / 2) {
+                arrayOfFigures[i] = figureSupplier.getRandomFigure();
+            } else {
+                arrayOfFigures[i] = figureSupplier.getDefaultFigure();
+            }
+        }
+        for (Figure figure : arrayOfFigures) {
+            figure.draw();
+        }

```

Comment:
you can draw a figure right after you've created it, eliminating the need to iterate over all the figures